src/EFCore/Metadata/Internal/InternalModelBuilder.cs

```diff
@@ -211,6 +212,243 @@ public InternalModelBuilder([NotNull] Model metadata)
+                        rightEntityType.Name,
+                        rightSkipNavigation.Name,
+                        associationEntityTypeName));
+            }
+
+            leftSkipNavigation.Builder.HasForeignKey(leftForeignKey, configurationSource);
+            rightSkipNavigation.Builder.HasForeignKey(rightForeignKey, configurationSource);
+
+            // Creating the primary key below also negates the need for an index on
+            // the properties of leftForeignKey - that index is automatically removed.
+            associationEntityTypeBuilder.PrimaryKey(
+                leftForeignKey.Properties.Concat(rightForeignKey.Properties).ToList(),
+                configurationSource);
+
+            return associationEntityTypeBuilder;
+        }
+
+        private static ForeignKey CreateSkipNavigationForeignKey(
+            SkipNavigation skipNavigation,
+            InternalEntityTypeBuilder associationEntityTypeBuilder,
+            ConfigurationSource configurationSource)
+        {
+            var principalEntityType = skipNavigation.DeclaringEntityType;
+            var principalKey = principalEntityType.FindPrimaryKey();
+            if (principalKey == null)
+            {
+                return null;
+            }
+
+            var dependentEndForeignKeyPropertyNames = new List<string>();
+            var otherIdentifiers = associationEntityTypeBuilder.Metadata
+                .GetDeclaredProperties().ToDictionary(p => p.Name, p => 0);
+            foreach (var property in principalKey.Properties)
+            {
+                var propertyName = Uniquifier.Uniquify(
+                    string.Format(
+                        AssociationPropertyNameTemplate,
+                        principalEntityType.ShortName(),
+                        property.Name),
+                    otherIdentifiers,
+                    int.MaxValue);
+                dependentEndForeignKeyPropertyNames.Add(propertyName);
+                otherIdentifiers.Add(propertyName, 0);
+            }
+
+            return associationEntityTypeBuilder
+                .HasRelationship(
+                    principalEntityType.Name,
+                    dependentEndForeignKeyPropertyNames,
+                    principalKey,
+                    configurationSource)
+                .IsUnique(false, configurationSource)
+                .Metadata;
+        }
+
+        /// <summary>
+        ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+        ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+        ///     any release. You should only use it directly in your code with extreme caution and knowing that
+        ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+        /// </summary>
+        public virtual InternalModelBuilder RemoveAssociationEntityIfAutomaticallyCreated(
```

Comment:
```suggestion
        public virtual InternalModelBuilder RemoveAssociationEntityIfCreatedImplicitly(
```
"Automatically" is too generic, same goes for `IsAutomaticallyCreatedAssociationEntityType`

Reply:
Why automatic, why not conventionally? 🙃

Reply:
"Conventionally" would be fine too. "Implicitly" sounds slightly better to me so that's what I've changed it to.